GoogleForCreators/web-stories-wp
# Settings: Replace old popover component with new context menu component

## Context

Integrating the new design system into the Dashboard.

## Summary

Adds the new context menu in 3 places (replacing the popover menu card):
- My Stories view
- Saved Templates view
- Settings view

## Relevant Technical Choices

- Give links the permission to have an `onClick` event handler. Clicking a link should allow for other functionality to occur other than navigation.
- Tried to split up the big event handler that was passed into the old popover menu card. Each item in the new context menu receives its own `onClick` prop. Left a comment in the PR on the code to chat about this further.

## To-do

There's a few tests failing. Need to fix those and then we'll be gucci.

- [x] fix failing js and karma tests

## User-facing changes



|View|Before|After|
|--|--|--|
|Settings|![settings-old](https://user-images.githubusercontent.com/22185279/108557092-422fad80-72b5-11eb-869c-9bdb262c7afb.gif)|![settings](https://user-images.githubusercontent.com/22185279/108556890-efee8c80-72b4-11eb-88de-1097cf168161.gif)|


## Testing Instructions



For each view test that the following works:
1. Opening the context menu.
2. Clicking away from the context menu should close it.
3. Clicking each item in the menus should do the action it used to do (nothing has functionally changed).

(Optional)Verify that the menus are keyboard accessible. This should have been tested in other prs.

### QA






This PR can be tested by following these steps:

For each view test that the following works:
1. Opening the context menu.
2. Clicking away from the context menu should close it.
3. Clicking each item in the menus should do the action it used to do (nothing has functionally changed).

(Optional)Verify that the menus are keyboard accessible. This should have been tested in other prs.

### UAT






## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Partially addresses #6102
